avocado/utils/data_structures.py

```diff
@@ -243,3 +245,58 @@ def time_to_seconds(time):
     else:
         seconds = 0
     return seconds
+
+
+class DataSize(object):
+    """
+    Data Size object with builtin unit-converted attributes.
+
+    :param data: Data size plus optional unit string. i.e. '10Mb'. No
+                 unit string means the data size is in bytes.
+    """
+
+    def __init__(self, data):
+        self.__multipliers = OrderedDict([('b', 1),  # 2**0
+                                          ('k', 1024),  # 2**10
+                                          ('m', 1048576),  # 2**20
+                                          ('g', 1073741824),  # 2**30
+                                          ('t', 1099511627776)])  # 2**40
+
+        pattern = r"([0-9]+)([a-z]+)?"  # Number and optional string
```

Comment:
We could improve this regex by adding start and end markers (`^` and `$`) and only allowing the units we can handle.

Reply:
Ok for `^` and `$`. But for the units, I went for flexibility here.   `100Mbytes`, `1GB`, `40bytes`, `2kilobytes`, they will all pass this regex. If the first character of the string part, in lower case, is not a key in `self.__multipliers`, then I raise the `AttributeError` below.

Reply:
Well, have you thought about the use of 'mebibytes' and the likes?  What is the value of `DataSize("1Mebibytes")`?

Reply:
hmm, indeed, I didn't. The same issue with `Megabits`... or `Megawats` :)
Ok, I'm buying it. Thanks!